### PR TITLE
implement schema providers for caseClass, jsonSchemaFile and xsdFile (#512)

### LINKED ISSUE #512: Define DataObject Schema by Scala CaseClass
**Is your feature request related to a problem? Please describe.**
For several DataObjects a schema can be specified. Currently this has to be done as Spark DDL. This is ok if you have no Java project, otherwise it would be nice to define the schema by a case class, as done for Spark Datasets.

**Describe the solution you'd like**
Allow to specify a case class name as schema instead of Spark DDL string. Use prefix "caseClass#" to differentiate from DDL string.

### PR DESCRIPTION
**Pull request by zzeekk**
### What changes are included in the pull request?
see #512

### CODE REVIEW

File: sdl-core/src/main/scala/io/smartdatalake/util/misc/SchemaUtil.scala [hunk 1]
```diff
@@ -87,18 +97,18 @@ object SchemaUtil {
    * @param ignoreNullability whether to ignore differences in nullability.
    * @return `true` iff `leftType` is a subset of `rightType`. `false` otherwise.
    */
-  private def deepIsTypeSubset(leftType: GenericDataType, rightType: GenericDataType, ignoreNullability: Boolean, caseSensitive: Boolean ): Boolean = {
-    if (leftType.typeName != rightType.typeName) false  /*fail fast*/
+  private def deepIsTypeSubset(leftType: GenericDataType, rightType: GenericDataType, ignoreNullability: Boolean, caseSensitive: Boolean): Boolean = {
```
Comment (pgruetter) [inline sdl-core/src/main/scala/io/smartdatalake/util/misc/SchemaUtil.scala:100]: In `schemaDiff`, the parameter is called `ignoreNullable`. Can you please use the same term here?

**Review by pgruetter — APPROVED** [review]
Nice feature, this saves us converting a given XSD/JSON schema to a DDL string 👍 
Added one comment, but looks good to me otherwise.

**Comment by zzeekk** [comment]
Improvements to do for xsdSchema:
- support empty RowTag
- RowTag must be given as XPath expression, but RowTag of SparkDataSource is just a tag. If the tag is present in different places of the schema, every occurance is read by SparkDataSource - this might create confusion if schemas of the different places is not the same...

**Comment by zzeekk** [comment]
see #548

File: sdl-core/src/main/scala/io/smartdatalake/util/misc/SchemaUtil.scala (hunk 1 above)
Comment (zzeekk) [reply sdl-core/src/main/scala/io/smartdatalake/util/misc/SchemaUtil.scala:100]: done

**Pull request closed without merging**